models/models.js

```diff
@@ -39,50 +39,50 @@ exports.selectArticles = (sort_by, order, topic) => {
         return Promise.reject({ status: 400, msg: 'Invalid order value' });
     }
 
+    return db.query('SELECT slug FROM topics;').then((data) => {
+            const topics = data.rows.map((obj) => {
+                return obj.slug
+        })
 
+        let queryStr = `SELECT 
+                            articles.author,
+                            articles.title,
+                            articles.article_id,
+                            articles.topic,
+                            articles.created_at,
+                            articles.votes,
+                            articles.article_img_url,
+                            COALESCE(comment_counts.comment_count, 0) AS comment_count
+                        FROM articles
+                        LEFT JOIN (
+                            SELECT article_id, COUNT(article_id) AS comment_count 
+                            FROM comments 
+                            GROUP BY article_id
+                        ) AS comment_counts
+                        ON articles.article_id = comment_counts.article_id`
+
+        if (new_topic !== 'everything'){
+            if (!topics.includes(new_topic.toLowerCase())) {
+                return Promise.reject({ status: 400, msg: 'Invalid topic value' });
+            }
 
-    let queryStr = `SELECT 
-                        articles.author,
-                        articles.title,
-                        articles.article_id,
-                        articles.topic,
-                        articles.created_at,
-                        articles.votes,
-                        articles.article_img_url,
-                        COALESCE(comment_counts.comment_count, 0) AS comment_count
-                    FROM articles
-                    LEFT JOIN (
-                        SELECT article_id, COUNT(article_id) AS comment_count 
-                        FROM comments 
-                        GROUP BY article_id
-                    ) AS comment_counts
-                    ON articles.article_id = comment_counts.article_id`
-
-    
-    
-    
-    
-    if (new_topic !== 'everything'){
-        if (!validTopicValues.includes(new_topic.toLowerCase())) {
-            return Promise.reject({ status: 400, msg: 'Invalid topic value' });
+            queryStr += ` WHERE articles.topic = '${new_topic}'`
         }
 
-        queryStr += ` WHERE articles.topic = '${new_topic}'`
+        queryStr += ` ORDER BY articles.${new_sort_by} ${new_order};`
 
-
-    }
-    queryStr += ` ORDER BY articles.${new_sort_by} ${new_order};`
-
-
-
-    return db.query(queryStr).then(({rows}) => {
-        return rows
+        return db.query(queryStr).then(({rows}) => {
+            return rows
+        })
```

Comment:
You may not want nested .then(). It's better to only interact with the db once in each model. Could check the topic existed or not in the controller